File: problems/p/provincesandgold/provincesandgold.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    g, s, c = [int(val) for val in input().split()]

    total = (g * 3) + (s * 2) + (c * 1)

    victory = None
    if total >= 8:
        victory = 'Province'
    elif total >= 5:
        victory = 'Duchy'
    elif total >= 2:
        victory = 'Estate'

    treasure = 'Copper'
    if total >= 6:
        treasure = 'Gold'
    elif total >= 3:
        treasure = 'Silver'
    if victory == None:
        print(treasure)
    else:
        print(f'{victory} or {treasure}')


if __name__ == '__main__':
    main()

File: problems/p/provincesandgold/provincesandgold.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <iostream>

using namespace std;

int main() {
    int g, s, c, total;

    cin >> g >> s >> c;

    total = (g * 3) + (s * 2) + (c * 1);

    if (total >= 8) {
        cout << "Province or ";
    }
    else if (total >= 5) {
        cout << "Duchy or ";
    }
    else if (total >= 2) {
        cout << "Estate or ";
    }

    if (total >= 6) {
        cout << "Gold" << endl;
    }
    else if (total >= 3) {
        cout << "Silver" << endl;
    }
    else {
        cout << "Copper" << endl;
    }
}